unstructured_ingest/v2/processes/connectors/outlook.py

```diff
@@ -0,0 +1,215 @@
+        "Use quotes if there are spaces in folder names."
+    )
+    recursive: bool = Field(
+        default=False,
+        description="Recursively download files in their respective folders otherwise stop at the"
+        " files in provided folder level.",
+    )
+    user_email: str = Field(description="Outlook email to download messages from.")
+
+
+@dataclass
+class OutlookIndexer(Indexer):
+    index_config: OutlookIndexerConfig
+    connection_config: OutlookConnectionConfig
+    connector_type: str = CONNECTOR_TYPE
+
+    def run(self, **kwargs: Any) -> Generator[FileData, None, None]:
+        messages = self._list_messages(recursive=self.index_config.recursive)
+
+        for message in messages:
+            yield self._message_to_file_data(message)
+
+    def run_async(self, **kwargs: Any) -> Coroutine[Any, Any, Any]:
+        raise NotImplementedError
+
+    def is_async(self) -> bool:
+        return False
+
+    def _list_messages(self, recursive: bool) -> list["Message"]:
+        mail_folders = self._get_selected_root_folders()
+        messages = []
+
+        while mail_folders:
+            mail_folder = mail_folders.pop()
+            messages += list(mail_folder.messages.get().top(MAX_EMAILS_PER_FOLDER).execute_query())
+
+            if recursive:
+                mail_folders += list(mail_folder.child_folders.get().execute_query())
+
+        return messages
+
+    def _get_selected_root_folders(self) -> list["MailFolder"]:
+        client_user = self.connection_config.get_client().users[self.index_config.user_email]
+        root_mail_folders = client_user.mail_folders.get().execute_query()
+
+        selected_names_normalized = [
+            folder_name.lower() for folder_name in self.index_config.outlook_folders
+        ]
+        selected_root_mail_folders = [
+            folder
+            for folder in root_mail_folders
+            if folder.display_name.lower() in selected_names_normalized
+        ]
+
+        if not selected_root_mail_folders:
+            logger.error(
+                f"Root folders selected in configuration: {self.index_config.outlook_folders}"
+                f"not found for user email {self.index_config.user_email}. Aborting."
+            )
+            raise ValueError("Root folders selected in configuration not found.")
+
+        return selected_root_mail_folders
+
+    def _message_to_file_data(self, message: "Message") -> FileData:
+        fullpath = self._generate_fullpath(message)
+
+        return FileData(
+            identifier=message.id,
+            connector_type=CONNECTOR_TYPE,
+            source_identifiers=SourceIdentifiers(filename=fullpath.name, fullpath=str(fullpath)),
+            metadata=FileDataSourceMetadata(
+                url=message.web_link,
+                # TODO(Filip Knefel): Based on V1 I used web_link but there's also
+                # message.resource_url, which one better suits this field?
```

Comment:
Copying over my answer for easier read, which is more akin to what this url in metadata is supposed to provide?
> - `resource_url` is a graph API url we'd send the request to, in order to access the resource e.g.
>   `https://graph.microsoft.com/v1.0/users/<email>onmicrosoft.com/messages/<message-id>`
> - `web_link` is a link to view the resource content via outlook in the browser

Reply:
Hmm it depends on what's the intent of url field in FileDataSourceMetadata. There's no docs or comment about it so hard to tell. But I would guess that it's for storing url so that the resource can be accessed/fetched by machine. So a `resource_url` matches that better. But if `web_link` was used already, then maybe it was just for having a identifier... uri let's say.

Reply:
I'm getting an error here on web_link. But resource_url works.

Reply:
Lets use `resource_url`